server/src/main/java/org/apache/iotdb/db/metadata/mtree/MTree.java

```diff
@@ -1767,6 +1767,13 @@ public int getMountedNodeIndexOnMeasurementPath(PartialPath measurementPath)
             : template.getRelatedStorageGroup();
     List<String> resSet = new ArrayList<>();
     for (PartialPath sgPath : initPath) {
+      // before Traverser refactored as 0.14, need check storage group alone
+      if (getNodeByPath(sgPath).getSchemaTemplate() != null) {
+        if (templateName.equals(ONE_LEVEL_PATH_WILDCARD)
+            || templateName.equals(getNodeByPath(sgPath).getSchemaTemplate().getName())) {
+          resSet.add(sgPath.getFullPath());
+        }
+      }
```

Comment:
According to this scenario, if you want to include the sg node, it is better to use ```traverser.setPrefixMatch(true)``` .

Reply:
Thanks a lot ! And I have fixed as you suggested